pkg/operators/otel-metrics/otel-metrics.go

```diff
@@ -675,6 +682,26 @@ func (m *otelMetricsOperatorInstance) PreStart(gadgetCtx operators.GadgetContext
 			hasValueFields = true
 		}
 
+		// Read additional static keys from annotations
+		staticValFns, err := helpers.GetKeyValueFuncsFromAnnotations[attribute.Key, attribute.Value](
+			annotations,
+			AnnotationKeysPrefix,
+			AnnotationKeysValueSuffix,
+			AnnotationKeysTypeSuffix,
+			attribute.Int64Value,
+			attribute.Float64Value,
+			attribute.StringValue,
+		)
+		if err != nil {
+			return fmt.Errorf("preparing static keys: %w", err)
+		}
+		for _, fn := range staticValFns {
+			k, v := fn()
```

Comment:
Why does `GetKeyValueFuncsFromAnnotations` need to return lambda functions if it's evaluated right after on this loop?